### PR TITLE
IODEMO: fix io message size and iov.calc_chksum()

### PR DESCRIPTION
**Pull request by evgeny-leksikov**

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
@yosefe  pls take a look

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -1125,7 +1171,7 @@ static int parse_args(int argc, char **argv, options_t *test_opts)
     test_opts->max_data_size        = 4096;
     test_opts->chunk_size           = std::numeric_limits<unsigned>::max();
     test_opts->num_offcache_buffers = 0;
-    test_opts->iomsg_size           = 256;
+    test_opts->iomsg_size           = sizeof(P2pDemoCommon::iomsg_t);
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc]: need to revert

File: test/apps/iodemo/io_demo.cc [hunk 2]
```diff
@@ -480,13 +505,28 @@ class P2pDemoCommon : public UcxContext {
...
         return pass;
     }
 
+    bool validate(const iomsg_t *msg) {
+        BufferIov *iov = _data_buffers_pool.get();
+
+        iov->init(opts().iomsg_size, _data_chunks_pool, 0, false);
+        iov->set_data(msg, opts().iomsg_size);
+
+        bool pass = validate(msg->tr.sn, *iov);
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc]: need to use common func for validate to avoid data copy to buffer
(move to another PR, this PR will not validate iomsg payload)

File: test/apps/iodemo/io_demo.cc [hunk 3]
```diff
@@ -341,12 +360,18 @@ class P2pDemoCommon : public UcxContext {
...
             if (validate) {
-                m->hdr.chk_sum = ucs_crc16(&m->tr, sizeof(*m) - sizeof(m->hdr));
+                const uint8_t *end = reinterpret_cast<uint8_t*>(_buffer) +
+                                     _io_msg_size;
+                uint8_t *tail      = reinterpret_cast<uint8_t*>(m + 1);
+                while (tail < end) {
+                    *(tail++) = IoDemoRandom::rand<uint8_t>();
+                }
+
+                m->hdr.chk_sum = ucs_crc16(&m->tr, _io_msg_size - sizeof(m->hdr));
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc]: move to next PR

File: test/apps/iodemo/io_demo.cc [hunk 4]
```diff
@@ -609,6 +619,12 @@ class DemoServer : public P2pDemoCommon {
                     << msg->tr.sn << " data size " << msg->data_size
                     << " conn " << conn;
 
+        if (opts().validate) {
+            if (!validate(msg)) {
+                abort();
+            };
+        }
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc]: maybe validate() should call abort itself (same for data validate)

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@evgeny-leksikov pls squash

**Comment by evgeny-leksikov** [comment]
> @evgeny-leksikov pls squash

done
